torchtune/data/_utils.py

```diff
@@ -42,18 +44,66 @@ def truncate(
     return tokens_truncated
 
 
-def split_text_by_image_tag(content: str, image_tag: str) -> List[Dict[str, str]]:
+def load_image(image_loc: Union[Path, str]) -> "PIL.Image.Image":
+    """
+    Convenience method to load an image in PIL format from a local file path or remote source.
+
+    Args:
+        image_loc (Union[Path, str]): Local file path or remote source pointing to the image
+            which will be loaded in PIL format.
+
+    Note:
+        If loading an image from a remote source, the function expects the URL provided in ``image_loc``
+        to start with "http" or "https" e.g. "https://www.wikipedia.org/en/bird.jpg".
+
+    Raises:
+        ValueError: If the image cannot be loaded from remote source.
+        ValueError: If the image cannot be opened as a PIL.Image.
+
+    Examples:
+        >>> # Load from remote source
+        >>> image = load_image("https://www.wikipedia.org/en/bird.jpg")
+
+        >>> # Load from local file path
+        >>> image = load_image(Path("/home/user/bird.jpg"))
+
+    Returns:
+        PIL.Image.Image: The loaded image.
+    """
+    # Hackily import PIL to avoid burdensome import in the main module
+    # TODO: Fix this
+    from PIL import Image
+
+    # If pointing to remote source, try to load to local
+    if isinstance(image_loc, str) and image_loc.startswith("http"):
+        try:
+            image_loc = request.urlopen(image_loc)
+        except Exception as e:
+            raise ValueError(f"Failed to load image from {image_loc}") from e
+
+    # Open the local image as a PIL image
+    try:
+        image = Image.open(image_loc)
+    except Exception as e:
+        raise ValueError(f"Failed to open image as PIL Image from {image_loc}") from e
```

Comment:
nit: are these really ValueErrors? (Or are you doing the thing I also do and lazily using `ValueError` as a catch-all for things that are not actually ValueErrors)

Reply:
WRT the first one, no it could be several different errors. Trying to call a remote URL always has plenty of exception types. I could just leave the error as is and let the user deal with it, but that seems messy.

WRT the second value error, more often than not, it will be a ValueError as the provided value is unable to be opened by the PIL Image interface. 

Do you want me to remove the exceptions with error messages and just let it ride all the way to the user?

Reply:
OK thanks for the explanation, in that case I think what you have is fine